clash-lib/src/Clash/Normalize/Transformations.hs

```diff
@@ -2472,6 +2482,61 @@ separateArguments ctx@(TransformContext is0 _) e@(Lam b eb0) = do
         isN1 = extendInScopeSet isN0 x'
     in  (isN1,x')
 
+  -- When splitting up a single argument into multiple arguments (see docs of
+  -- 'separateArguments') we should make sure to update TopEntity annotations
+  -- accordingly. See: https://github.com/clash-lang/clash-compiler/issues/1033
+  splitTopAnn
+    :: HasCallStack
+    => Int
+    -- ^ Number of ports to insert
+    -> (TopEntityT, PortName)
+    -- ^ TopEntity to insert ports in / port name of
+    -> NormalizeSession ()
+  splitTopAnn n (t0@(TopEntityT id_ (Just s@(Synthesize{t_inputs})) _tb), portName) =
+    case portName of
+      PortProduct nm portNames0 ->
+        let
+          (pre, tail -> post) = splitAt (length cctx) t_inputs
+          newPortNames = [PortName (nm ++ "_" ++ show i_) | i_ <- [(0::Int)..]]
+          portNames1 = take n (portNames0 ++ newPortNames)
+          t1 = t0{topAnnotation=Just s{t_inputs=pre ++ portNames1 ++ post}}
+        in do
+          (cf, _) <- Lens.use curFun
+          topEntities Lens.%= extendVarEnv cf t1
```

Comment:
This fails to update `portNames0`. `nm` + `_` should be prepended to all `PortName`s / `PortProduct`s in `portNames0`.

Reply:
Fixed.